lxd/daemon.go

```diff
@@ -302,6 +302,73 @@ func allowPermission(entityType entity.Type, entitlement auth.Entitlement, muxVa
 	}
 }
 
+// allowProjectResourceList should be used instead of allowAuthenticated when listing resources within a project.
+// This prevents a restricted TLS client from listing resources in a project that they do not have access to.
+func allowProjectResourceList(d *Daemon, r *http.Request) response.Response {
+	// The caller must be authenticated.
+	if !auth.IsTrusted(r.Context()) {
+		return response.Forbidden(nil)
+	}
+
+	isRoot, err := auth.IsRootUserFromCtx(r.Context())
+	if err != nil {
+		return response.InternalError(fmt.Errorf("Failed to determine caller privilege: %w", err))
+	}
+
+	// A root user can list resources in any project.
+	if isRoot {
+		return response.EmptySyncResponse
+	}
+
+	authenticationMethod, err := auth.GetAuthenticationMethodFromCtx(r.Context())
+	if err != nil {
+		return response.InternalError(fmt.Errorf("Failed to determine caller authentication method: %w", err))
+	}
+
+	// OIDC authenticated clients are governed by fine-grained auth. They can call the endpoint but may see an empty list.
+	if authenticationMethod == api.AuthenticationMethodOIDC {
+		return response.EmptySyncResponse
+	}
+
+	username, err := auth.GetUsernameFromCtx(r.Context())
+	if err != nil {
+		return response.InternalError(fmt.Errorf("Failed to determine caller username: %w", err))
+	}
+
+	id, err := d.identityCache.Get(authenticationMethod, username)
```

Comment:
For potential future work, would it make sense to move this logic into the IsRootUser function?

Also i feel hasAdminAccess would be more accurate, as we dont really mentuim the concept of a  "root" user in lxd docs.

Reply:
Perhaps `IsServerAdmin`? As this aligns with the `admin` relation on `server` in the auth model. I think it does make sense to move it. I think I can do this fairly quickly so I'll put up a quick PR.